include/pybind11/pybind11.h

```diff
@@ -1344,15 +1344,27 @@ PYBIND11_NOINLINE inline void print(tuple args, dict kwargs) {
     auto sep = kwargs.contains("sep") ? kwargs["sep"] : cast(" ");
     auto line = sep.attr("join")(strings);
 
-    auto file = kwargs.contains("file") ? kwargs["file"].cast<object>()
-                                        : module::import("sys").attr("stdout");
+    object file;
+    if (kwargs.contains("file")) {
+        file = kwargs["file"].cast<object>();
+    } else {
+        try {
+            file = module::import("sys").attr("stdout");
+        } catch (const import_error &) {
+            /* If print() is called from code that is executed as
```

Comment:
It came up when I added ``py::print`` to a destructor that (as part of implementing a new testcase) got called during interpreter shutdown. Many Python calls will still work in that situation, but importing new things is definitely forbidden. ``py::print`` is special in this regard that it needs to import ``sys``, and then ``stdout``, hence the decision to modify it.